### ISSUE TITLE
App freezes on UI thread for the first time of ORM intiliazation for a long time

### ISSUE THREAD

**Issue by NorbertSzydlowski** [satyan/sugar#212]
App freezes on UI thread for the first time of ORM intiliazation for a long time,
any solution for this?

**Comment by perfectmak** [comment]
Yes, I'm also experiencing this in my application. Any solution for this

**Comment by svenoaks** [comment]
I have this too, I'm going to look at a way to create the database upfront, at a more opportune time. Will update on any good solution.

**Comment by jmate003** [comment]
Has anyone found a solution for this?

**Comment by SoAddicted** [comment]
Haven't checked it yet, but maybe with this commit: https://github.com/satyan/sugar/commit/bcb9badc79aa3149f847e72efbe9da3d09923947

you guys can init in background thread?

**Comment by akoul889** [comment]
Any Update on This??

**Comment by j-mateo** [comment]
What I did in order to fix this was to make sure that my first query to the database was in Async. After doing that, it appears that the initialization happens in the background. I extended SugarRecord and added functions to make all the db transactions in a background thread. For example, Record.listAll(callback)

**Comment by akoul889** [comment]
@j-mateo Thanks..will try this.
